feature-currency/src/test/java/com/stefang/app/feature/currency/viewmodel/HistoryConversionViewModelTestWithMockk.kt

```diff
@@ -0,0 +1,103 @@
+package com.stefang.app.feature.currency.viewmodel
+
+import app.cash.turbine.test
+import app.cash.turbine.turbineScope
+import com.stefang.app.core.data.log.Logger
+import com.stefang.app.core.data.repository.HistoryRepository
+import com.stefang.app.core.testing.repository.TestHistoryRepository
+import com.stefang.app.core.testing.util.MainDispatcherRule
+import com.stefang.app.feature.currency.model.HistoryConversionUiModel
+import com.stefang.app.feature.currency.viewmodel.HistoryConversionViewModel.SnackBarEvent.DeleteAll
+import com.stefang.app.feature.currency.viewmodel.HistoryConversionViewModel.SnackBarEvent.DeleteSingle
+import io.mockk.MockKAnnotations
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertIs
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class HistoryConversionViewModelTestWithMockk {
+
+    @get:Rule
+    val dispatcherRule = MainDispatcherRule()
+
+    @RelaxedMockK
+    private lateinit var historyRepository: HistoryRepository
+
+    @RelaxedMockK
+    private lateinit var logger: Logger
+
+    private lateinit var viewModel: HistoryConversionViewModel
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
+
+        every { historyRepository.histories } returns flowOf(TestHistoryRepository.histories)
+        viewModel = HistoryConversionViewModel(historyRepository, logger)
+    }
+
+    @Test
+    fun `when historyRepository returns histories then allHistories should return the same list`() = runTest {
+        val collectJob = launch(dispatcherRule.testDispatcher) {
+            viewModel.allHistories.collect()
+        }
+        // it won't work if we stub the flow here, because viewModel.allHistories is initialized
+        // once the viewModel is created, so we need to stub the flow before creating the viewModel (setUp())
+//        every { historyRepository.histories } returns flowOf(TestHistoryRepository.histories)
```

Comment:
Test nya ku bikin 2 versi, 1: Pake mockk, 2: Pake Dummy class.
Yang pke mockk ini ada issue di testing Flow ini

Reply:
What's the possibilities here Mas?